pgerror: add severity concept to pgerror package

otan opened the pull request:
`pgwire` will need to send down severities which  can be customised by
the developer.

To accomplish this, we introduce severity to the pgerror package, which
includes the helper functions `WithSeverity` and `GetSeverity`.
Developers who specify a severity using WithSeverity will now see that
severity pop up over the wire. The default severity if none is specified
is ERROR.

This has been plugged in for NOTICE using `Noticef`.

Refs: https://github.com/cockroachdb/cockroach/issues/44711

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47630)
<!-- Reviewable:end -->

otan committed:
pgerror: add severity concept to pgerror package

`pgwire` will need to send down severities which  can be customised by
the developer.

To accomplish this, we introduce severity to the pgerror package, which
includes the helper functions `WithSeverity` and `GetSeverity`.
Developers who specify a severity using WithSeverity will now see that
severity pop up over the wire. The default severity if none is specified
is ERROR.

This has been plugged in for NOTICE using `Noticef`.

Release note: None

knz reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> very nice!!

> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47630#-:-M5CBu6-8aMIYFdLdzX1:b-l0qzxk)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=knz
